### PR TITLE
Send tokens back to owner chain when orders are filled or cancelled.

### PR DESCRIPTION
**Pull request by afck**
## Motivation

Currently the matching engine keeps all tokens on the order book chain. Owners have to claim them to get them back to their own chains.

This means the chain can't be used for atomic swaps: When closing the chain, the matching engine application does not have permissions to move those tokens back to their owners' chains. Users may also find it surprising or inconvenient in general.

## Proposal

Whenever an order is filled or cancelled, send the tokens to the owner's chain.

## Test Plan

The integration test was updated.

## Links

- #305 
- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

**Review by andresilva91 — APPROVED** [review]
Tests seem to be failing with code unrelated to this PR? `chrono::NaiveDateTime::from_timestamp_opt` seems to have been deprecated. Accepting to unblock, will leave it up to you to fix this here or in a separate PR (before this one)

**Comment by afck** [comment]
Yes, fixed in https://github.com/linera-io/linera-protocol/pull/1765.

**Pull request merged**
